packages/langium/src/lsp/language-server.ts

```diff
@@ -171,9 +183,28 @@ export class DefaultLanguageServer implements LanguageServer {
     }
 
     async initialized(params: InitializedParams): Promise<void> {
+        await this.fireInitializedOnDefaultServices(params);
+
         this.onInitializedEmitter.fire(params);
         this.onInitializedEmitter.dispose();
     }
+
+    protected fireInitializeOnDefaultServices(params: InitializeParams): void {
+        this.services.workspace.ConfigurationProvider.initialize(params);
+        this.services.workspace.WorkspaceManager.initialize(params);
+    }
```

Comment:
True, but then the constructors would have to rely on `services.lsp.LanguageServer....`, and that's not possible if the LSP part is not available. And we didn't want to have a core workspace manager and an LSP-empowered specialized one.

Reply:
> And we didn't want to have a core workspace manager and an LSP-empowered specialized one.

I probably didn't see the discussion on this. Can you link it? I'm personally in favor of having a dedicated LSP workspace service, since this introduces a lot of unexpected pollution in the default workspace manager, i.e. the `initialize` message, which has an argument type coming from the `vscode-languageserve-types` package.

Reply:
*sigh*, alright, I guess I'm fine with this. We should write some custom eslint rules to prevent people from accidentally importing things they are not allowed to import.